Assets/Character/Scripts/Consumable.cs

```diff
@@ -0,0 +1,49 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class Consumable : MonoBehaviour {
+
+    private int count;
+    private float cooldown;
+    private float startTime;
+    private bool healing;
+    private float duration;
+    private float interval;
+    private float timer;
+
+	// public Consumable?
+	void Start () {
+        count = Constants.CONSUMABLE_COUNT;
+        cooldown = Constants.CONSUMABLE_CD;
+        healing = false;
+        duration = Constants.CONSUMABLE_DURATION;
+        timer = 0;
+	}
+	
+	void Update () {
+        if(healing)
+        {
+            if (Time.time - startTime > duration)
+            {
+                healing = false;
+            }
+            else
+            {
+                timer += Time.deltaTime;
+                if (timer > interval)
+                {
+                    //add hp
```

Comment:
add an abstract method for derived classes to implement. 

abstract void consume() //i don't remember the syntax

healthPotion.consume() //heals
staminaPotion.consume() //recovers stamina